libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -33,6 +33,14 @@ const AP_Param::GroupInfo AP_Mission::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO("OPTIONS",  2, AP_Mission, _options, AP_MISSION_OPTIONS_DEFAULT),
 
+    // @Param: MAX_WP_DIST
+    // @DisplayName: Maximum allowed distance to first waypoint.
+    // @Description: Maximum allowed distance in meters to first waypoint for a pre-arm check. Prevents a fly away. Set param to 0, to allow any distance.
+    // @Units: m
+    // @Range: 0 100000
+    // @User: Advanced
+    AP_GROUPINFO("MAX_WP_DIST",   3,  AP_Mission, _max_nav_dist_m, 0),
```

Comment:
We should defualt to some sensible number here since the user has to also go and turn on the check in the bitmask. Or we could default to 0 here for disabled and not bother adding it to the bitmask. 

I think I would vote to leave it out of the bitmask.

Reply:
This makes sense. A sensible value might be hard to find, between copter and plane. Leaving it out of the bitmask has the downside of spreading out where you disable mission checks, and it would change the `if` in `AP_Arming::mission_checks`. I don't know which would be more adequate.

Reply:
No reason it has to be the same default on plane and copter, you can `#define` it based on vehicle type. Although I don't think vehicle type makes a huge difference it is just a requirement on how you plan your mission.